### PR TITLE
micro new [service]

### PR DESCRIPTION
**Pull request by asim**
In relation to https://github.com/micro/micro/issues/41. This is a first crack at basically just generating some example boiler plate template. It means not having to copy and paste crap.

### CODE REVIEW

File: new/new.go [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+	// check if dir exists
+	if _, err := os.Stat(c.GoDir); !os.IsNotExist(err) {
+		return fmt.Errorf("%s already exists", c.GoDir)
+	}
+
+	fmt.Println("creating service", c.FQDN)
+
+	// write the files
+	for _, file := range c.Files {
+		f := path.Join(c.GoDir, file.Path)
```
Comment (HeavyHorst) [inline new/new.go]: to be platform independent we should use filepath.Join(c.GoDir, file.Path) to create filepaths.

File: new/new.go [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+	goPath := os.Getenv("GOPATH")
+
+	// don't know GOPATH, runaway....
+	if len(goPath) == 0 {
+		fmt.Println("unknown GOPATH")
+		return
+	}
+
+	goPath = strings.Split(goPath, ":")[0]
+	goDir := path.Join(goPath, "src", path.Clean(dir))
```
Comment (HeavyHorst) [inline new/new.go]: here also

File: new/new.go [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+	parts := strings.Split(dir, "/")
+
+	if len(alias) == 0 {
+		// set as last part
+		alias = parts[len(parts)-1]
```
Comment (HeavyHorst) [inline new/new.go]: alias = filepath.Base(dir) should work

**Comment by asim** [comment]
Looks like I've got api, srv and web template generation examples. Might be time to merge and close.

**Pull request merged**